submissions/shramkoweb/friends-app/script.js

```diff
@@ -0,0 +1,168 @@
+
+const createElement = (template) => {
+    const newElement = document.createElement('div');
+    newElement.innerHTML = template;
+    return newElement.firstElementChild;
+};
+
+const handleResetClick = () => {
+    state.search = '';
+    state.age = null;
+    state.name = null;
+    state.gender = Gender.ALL;
+
+    render(state);
+};
+
+const filterByGender = (users, gender) => {
+    if (gender === Gender.ALL) {
+        return users;
+    }
+
+    return users.filter(user => user.gender === gender);
+};
+
+const filterByQuery = (users, query) => {
+    if (query === '') {
+        return users;
+    }
+
+    return users.filter(
+        user => user.name.first
+            .toLowerCase()
+            .includes(query)
+    );
+};
+
+const sortByName = (users, sortType) => {
+    switch (sortType) {
+        case 'a':
+            return users.sort((a, b) => {
+                const firstNAme = a.name.first.toUpperCase();
+                const secondName = b.name.first.toUpperCase();
+
+                if (firstNAme < secondName) {
+                    return -1;
+                }
+                if (firstNAme > secondName) {
+                    return 1;
+                }
+
+                return 0;
+            });
+        case 'z':
+            return users.sort((a, b) => {
+                const firstNAme = a.name.first.toUpperCase();
+                const secondName = b.name.first.toUpperCase();
+
+                if (firstNAme < secondName) {
+                    return 1;
+                }
+                if (firstNAme > secondName) {
+                    return -1;
+                }
+
+                return 0;
+            });
+        default:
+            return users;
+    }
+};
+
+const sortBy = (users, sortType) => {
+    switch (sortType) {
+        case SortType.ASCENDING:
+            return users.sort((a, b) => a.dob.age - b.dob.age);
+        case SortType.DESCENDING:
+            return users.sort((a, b) => b.dob.age - a.dob.age);
+        default:
+            return users;
+    }
+};
+
+const renderCard = (data) => {
+    const { name, dob, gender, email, picture } = data;
+
+    const markup = `
+        <li class="friends__item">
+          <section class="friend-card">
+            <div class="friend-card__image-wrapper">
+              <img class="friend-card__image" src=${picture.large} alt=${name.first}>
+            </div>
+            <h3 class="friend-card__title">${name.first} ${name.last}</h3>
+            <a class="friend-card__mail" href="mailto:${email}">${email}</a>        
+            <p class="friend-card__sex">Gender - ${gender}</p>
+            <p class="friend-card__age">Age - ${dob.age}</p>
+          </section>
+        </li>
+    `;
+
+    return createElement(markup);
+};
+
+const renderCards = (cards, target) => {
+    const fragment = document.createDocumentFragment();
+    cards.forEach(card => fragment.appendChild(renderCard(card)));
```

Comment:
Your approach of adding card content is ok. Also, I appreciate you using `document.createDocumentFragment();`
Maybe I'd recommend a small improvement of your function `renderCard` because I was confused a little bit about some manipulations in the `createElement` function, where you created new nodes and took `firstElementChild` from there. 
As I know, JS parses strings faster than DOM nodes, so you can return only string content in the `renderCard` function, and then you can just concat your strings in `forEach` loop and then add this content to your fragment.
```suggestion
    let cardsContent = '';
    cards.forEach(card => cardsContent += renderCard(card));
    fragment.innerHTML = cardsContent;
    target.appendChild(fragment);
```

Reply:
Cool. But it seems to me that string concatenation will confuse even more, as for me =) But I will know about this performance trick)